### ISSUE TITLE
What does the size of sequence_examples depend on, and how to set its size?

### ISSUE THREAD

**Issue by 18liumin** [google/ml-compiler-opt#326]
def _sequence_example_dataset_fn(sequence_examples):
    # Data collector returns empty strings for corner cases, filter them out
    # here.
    # yapf: disable - Looks better hand formatted
    dataset = (tf.data.Dataset
                .from_tensor_slices(sequence_examples)
                .filter(lambda string: tf.strings.length(string) > 0)
                .map(parser_fn)
                .filter(lambda traj: tf.size(traj.reward) > 2)
                .unbatch()
               )
    # yapf: enable
    return dataset

**Comment by mtrofin** [comment]
This part of the code is quite convoluted (and not sure it needs to be *that* convoluted). IIRC at this point it's the concatenation of sequence examples that came from the various compilations.

If you set up a small example with the inliner (2 small modules that should produce different lengths of inlining decision sequences) and debug thorough, it should become clear. 

(of course, if you have an idea how to simplify the code, patches are welcome)

**Comment by 18liumin** [comment]
> If you set up a small example with the inliner (2 small modules that should produce different lengths of inlining decision sequences) and debug thorough, it should become clear.

Why should sequence_examples be set to 16? Can it be set to other sizes? What does this parameter depend on?

**Comment by 18liumin** [comment]
> This part of the code is quite convoluted (and not sure it needs to be _that_ convoluted). IIRC at this point it's the concatenation of sequence examples that came from the various compilations.
> 
> If you set up a small example with the inliner (2 small modules that should produce different lengths of inlining decision sequences) and debug thorough, it should become clear.
> 
> (of course, if you have an idea how to simplify the code, patches are welcome)

  def _sequence_example_dataset_fn(sequence_examples):
    # Data collector returns empty strings for corner cases, filter them out
    # here.
    # yapf: disable - Looks better hand formatted
    dataset = flat_sequence_example_dataset_fn(sequence_examples)
    return (dataset.batch(train_sequence_length, drop_remainder=True)
                   .cache()
                   .shuffle(trajectory_shuffle_buffer_size)
                   .batch(batch_size, drop_remainder=True))

  return _sequence_example_dataset_fn

Sorry, what I want to ask is how to set the size of train_sequence_length and why it is set to 16

**Comment by mtrofin** [comment]
I'm assuming it's this definition [here](https://github.com/google/ml-compiler-opt/blob/0f895e325491793efbc45a1675f9a71f7c8c4f9e/compiler_opt/rl/data_reader.py#L147) - there are a few others (btw, can you please use the permalink feature in the github web UI rather than code snippets, it's much easier to find the reference that way and make sure we are looking at the same thing)

You probably noticed how this is is passed from [here](https://github.com/google/ml-compiler-opt/blob/0f895e325491793efbc45a1675f9a71f7c8c4f9e/compiler_opt/rl/train_locally.py#L70). Note the `@gin.configurable` above `train_eval`. See [gin-config](https://github.com/google/gin-config). In our case, that comes from the `--gin_<...>`flags you pass in when you start training - `--gin_bindings` and `--gin_files`.

Assuming inlining, you probably used [this gin file](https://github.com/google/ml-compiler-opt/blob/0f895e325491793efbc45a1675f9a71f7c8c4f9e/compiler_opt/rl/inlining/gin_configs/ppo_nn_agent.gin#L18).

You can set it 2 ways:
- change it in the file. That's a good idea if you want to persist your settings
- pass `--gin_bindings=train_eval.train_sequence_length=1234` at the command line

**Comment by 18liumin** [comment]
> * train_sequence_length

what is the impact of setting train sequence_length bigger or smaller?

**Comment by 18liumin** [comment]
> I'm assuming it's this definition [here](https://github.com/google/ml-compiler-opt/blob/0f895e325491793efbc45a1675f9a71f7c8c4f9e/compiler_opt/rl/data_reader.py#L147) - there are a few others (btw, can you please use the permalink feature in the github web UI rather than code snippets, it's much easier to find the reference that way and make sure we are looking at the same thing)
> 
> You probably noticed how this is is passed from [here](https://github.com/google/ml-compiler-opt/blob/0f895e325491793efbc45a1675f9a71f7c8c4f9e/compiler_opt/rl/train_locally.py#L70). Note the `@gin.configurable` above `train_eval`. See [gin-config](https://github.com/google/gin-config). In our case, that comes from the `--gin_<...>`flags you pass in when you start training - `--gin_bindings` and `--gin_files`.
> 
> Assuming inlining, you probably used [this gin file](https://github.com/google/ml-compiler-opt/blob/0f895e325491793efbc45a1675f9a71f7c8c4f9e/compiler_opt/rl/inlining/gin_configs/ppo_nn_agent.gin#L18).
> 
> You can set it 2 ways:
> 
> * change it in the file. That's a good idea if you want to persist your settings
> * pass `--gin_bindings=train_eval.train_sequence_length=1234` at the command line

what is the impact of setting train sequence_length bigger or smaller?

**Comment by mtrofin** [comment]
Probably affects training time (larger/smaller datasets), but past that, best to experiment with a few values (i.e. do a hyperparameter sweep).
